pdl_migration.md

```diff
@@ -0,0 +1,414 @@
+---
+layout: guide
+title: Migrating from PDSC to PDL
+permalink: /pdl_migration
+excerpt: Guide for migrating from PDSC to PDL with a side-by-side comparison showing notable differences.
+---
+
+# Migrating from PDSC to PDL
+
+{{ page.excerpt }}
+
+## Contents
+- [Why Use PDL?](#why-use-pdl)
+- [How to Convert Schemas to PDL](#how-to-convert-schemas-to-pdl)
+- [Notable Differences Between PDSC and PDL](#notable-differences-between-pdsc-and-pdl)
+    - [PDL is More Java-Like](#pdl-is-more-java-like)
+    - [Import Statements](#import-statements)
+    - [Shorthand for Custom Properties](#shorthand-for-custom-properties)
+    - [Cleaner Enum Declarations](#cleaner-enum-declarations)
+    - [Specifying Default Field Values](#specifying-default-field-values)
+
+## Why Use PDL?
+
+PDL as a schema definition language was designed to be read and written by humans, making it a much more enjoyable experience for developers
+to define their service's models. PDSC, despite its syntax being simply a subset of JSON, cannot boast this human-readability factor.
+
+In addition to its inherently more readable syntax, PDL also features some extra shorthand which developers can leverage to write less and
+make their schemas more readable for others. See [below](#notable-differences-between-pdsc-and-pdl) for more information on this.
+
+## How to Convert Schemas to PDL
+
+If you have an existing project with PDSC schemas, converting everything to PDL is pretty straightforward.
+Rest.li's Gradle plugin provides a task `convert<sourceSet>ToPdl` which will automatically convert your schemas for you.
+It should be noted that this task verifies the converted schemas against the original schemas.
+If this verification fails, then the whole conversion will be aborted.
+
+It's suggested to use at least version `28.1.3` when running this task.
+
+The following command will convert every PDSC schema in your project to PDL.
+
+```
+gradle convertToPdl
+```
+
+You can selectively migrate only one particular module of your project with the following:
+
+```
+gradle :<moduleName>:convertToPdl
+```
+
+This task also takes in a few options:
+
+```
+gradle :<moduleName>:convertToPdl \
+    [-PconvertToPdl.reverse=(true|false)] \
+    [-PconvertToPdl.keepOriginal=(true|false)] \
+    [-PconvertToPdl.preserveSrcCmd]
+```
+
+| Property | Type | Description
+|----------|------|-------------
+| `convertToPdl.reverse` | boolean | If true, converts PDL schemas back to PDSC (and vice versa if false).
```

Comment:
Include skip-verification flag

Reply:
I thought that too at first when I wrote this, but the `convertToPdl` Gradle task is separate from the `SchemaFormatTranslator` class which supports this flag.